projects/plugins/jetpack/modules/sharedaddy/services/class-jetpack-mastodon-modal.php

```diff
@@ -0,0 +1,195 @@
+<defs>
+<linearGradient id="paint0_linear_89_11" x1="30.5" y1="0.0130005" x2="30.5" y2="65.013" gradientUnits="userSpaceOnUse">
+<stop stop-color="#6364FF"></stop>
+<stop offset="1" stop-color="#563ACC"></stop>
+</linearGradient>
+</defs>
+</svg>';
+
+	/**
+	 * Decide whether the modal should be displayed.
+	 * We want to show it when the user clicks on the Mastodon share button,
+	 * thus loading a page with ?share=mastodon&nb=1.
+	 *
+	 * @return bool
+	 */
+	public static function should_display_modal() {
+		if (
+			// phpcs:disable WordPress.Security.NonceVerification.Recommended -- we only check if the query strings exist, we do not use them.
```

Comment:
Not really a good reason. https://jetpackcrew.wordpress.com/2022/03/02/when-can-you-ignore-wordpress-security-nonceverification/#we-re-not-using-the-submitted-data-just-the-fact-that-a-field-is-set

The good reason is that we only use this to display a UI, not to change anything on the site.

Reply:
That's true. 3eb672f9a370f98fe8340d1b807ee7633555956c